e3sm_diags/driver/polar_driver.py

```diff
@@ -1,264 +1,400 @@
-                        # Regrid towards the lower resolution of the two
-                        # variables for calculating the difference.
-                        mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
-                            mv1_domain,
-                            mv2_domain,
-                            parameter.regrid_tool,
-                            parameter.regrid_method,
-                        )
-
-                        # Plotting
-                        diff = mv1_reg - mv2_reg
-                        metrics_dict = create_metrics(
-                            mv2_domain, mv1_domain, mv2_reg, mv1_reg, diff
-                        )
-
-                        parameter.var_region = region
-                        plot(
-                            parameter.current_set,
-                            mv2_domain,
-                            mv1_domain,
-                            diff,
-                            metrics_dict,
-                            parameter,
-                        )
-                        utils.general.save_ncfiles(
-                            parameter.current_set,
-                            mv1_domain,
-                            mv2_domain,
-                            diff,
-                            parameter,
-                        )
-
-            # For variables without a z-axis.
-            elif mv1.getLevel() is None and mv2.getLevel() is None:
-                for region in regions:
-                    logger.info("Selected region: {}".format(region))
-
-                    mv1_domain = utils.general.select_region(
-                        region, mv1, land_frac, ocean_frac, parameter
-                    )
-                    mv2_domain = utils.general.select_region(
-                        region, mv2, land_frac, ocean_frac, parameter
-                    )
-
-                    parameter.output_file = "-".join([ref_name, var, season, region])
-                    parameter.main_title = str(" ".join([var, season, region]))
-
-                    # Regrid towards the lower resolution of the two
-                    # variables for calculating the difference.
-                    mv1_reg, mv2_reg = utils.general.regrid_to_lower_res(
-                        mv1_domain,
-                        mv2_domain,
-                        parameter.regrid_tool,
-                        parameter.regrid_method,
-                    )
-
-                    # Special case.
-                    if var == "TREFHT_LAND" or var == "SST":
-                        if ref_name == "WILLMOTT":
-                            mv2_reg = MV2.masked_where(
-                                mv2_reg == mv2_reg.fill_value, mv2_reg
-                            )
-
-                        land_mask = MV2.logical_or(mv1_reg.mask, mv2_reg.mask)
-                        mv1_reg = MV2.masked_where(land_mask, mv1_reg)
-                        mv2_reg = MV2.masked_where(land_mask, mv2_reg)
```

Comment:
TODO: unclear what this corresponds to. (Still haven't removed this block).

Reply:
Can be removed because WILLMOTT is no longer used. Can also affect other branches

Reply:
Can the whole thing (i.e. the `if var == "TREFHT_LAND" or var == "SST":` block)  be removed? @chengzhuzhang specifically mentioned L228-231 (i.e., the `if ref_name == "WILLMOTT":` block)

Reply:
Thanks for checking. I think we can remove the whole block from 226 to 235.